company post 관련 API 추가/수정

SangYunLeee opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)

- [x] 기능 추가
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표
- 회사 게시글 CRUD 작성 및 게시글  임시저장 API 버그 수정

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)

1. 회사 소개 글 작성 API 구현
- 회사 소개 글을 추가/변경 가능하도록 구현

2. 회사 소개 임시 저장 글 API  버그 수정
  - 임시 저장 글 수정 시, 파일 관련 내용을 수정하지 않아도 파일 URL 이 변경되는 현상 수정
  - 임시 저장 글 수정 시, 업로드하는 파일 프로퍼티에 빈칸으로 전달 시에는 기존에 저장되어있는 파일이 삭제되도록 수정

3. 회사의 멤버만이 게시글을 수정할 수 있도록 수정

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)

- 게시글 추가/변경 시 파일을 업로드할 경우 DB와 파일을 둘 다 업데이트해줘야하는데 두가지를 모두 고려해서 만드는 것이 어려웠습니다.

<br />

SangYunLeee committed:
Add: 회사 게시글 API 추가
- 회사 게시글 CRUD API 구현

SangYunLeee committed:
Add: 임시 게시글 수정 시 게시글의 이전 파일(info, img)이 제거되도록 수정

SangYunLeee committed:
fix: 변경을 요청한 값만 Update 되도록 수정
- 원인: 프론트에서 게시글 임시저장 요청 시 변경을 요청하지 않은 값은 빈값으로 업데이트되는 현상 발생
- 해결: 요청하지 않은 값은 그대로 두도록 수정

SangYunLeee committed:
Add: DB에 유저 계정 추가 및 멤버 그룹 추가

SangYunLeee committed:
Add: 회사 게시글 추가화면 첫 진입 시 회사정보를 제공하도록 수정
- 원인: 회사 게시글 추가화면 첫 진입 시 이전에 저장된 회사게시글 정보가 없으면 아무 정보도 제공하지 않아 프론트에서 에러가 발생함.
- 해결: 첫 진입 시 회사 정보를 제공하여 프론트에서 첫 기본 텍스트값을 얻을 수 있도록 수정

SangYunLeee committed:
style: indent를 tab 대신 space 로 사용하도록 수정
- 원인: 코드 리뷰 시 가독성이 안좋음.
- 해결: tab 대신 space 로 사용토록 수정

SangYunLeee committed:
Add: 회사의 멤버만이 게시글을 수정할 수 있도록 수정

wooryeong commented on src/services/post.service.ts:
```diff
@@ -0,0 +1,79 @@
+
+  // 이미 작성 폼이 존재하는 지 확인
+  const postInfo = await postDao.getPost({usersId: postInput.usersId, companiesId: postInput.companiesId});
+  // TODO
+    // 유저가 입력하려는 회사의 대표멥버인지 확인
+    // 미들웨어 부분에서 처리해야되는 건지 고민.
```
불필요한 주석은 지우는게 좋을 듯 합니다  <(^-^)/ ~🌿

SangYunLeee replied:
감사합니다 ^_^ 리뷰 반영하였읍니다 ~!

wooryeong commented on src/services/postForm.service.ts:
```diff
@@ -27,29 +39,50 @@ const putPostForm = async (postFormInput: CompanyPostFormInput) => {
-  const postFormInfo = await postDao.getPostForm({usersId: postFormInput.usersId, companiesId: postFormInput.companiesId});
+  const postFormInfo = await postFormDao.getPostForm({usersId: postFormInput.usersId, companiesId: postFormInput.companiesId});
 
   // 유저가 입력하려는 회사의 대표멥버인지 확인
     // TODO
   // 존재하면 업데이트
```
여기 주석도 확인 부탁드립니다아~ ^---^

SangYunLeee replied:
수정했읍니다 ㅎ.ㅎ ~!

wooryeong reviewed (approved):
완전 고생하신게 눈에 보이네염 ㅠ ㅠ 고생하셨습니다 🥲

SangYunLeee committed:
fix: 변경을 요청한 값만 Update 되도록 수정
 - 원인: 프론트에서 게시글 저장 요청 시 변경을 요청하지 않은 값은 빈값으로 업데이트되는 현상 발생
 - 해결: 요청하지 않은 값은 그대로 두도록 수정

SangYunLeee committed:
style: indent를 tab 대신 space 로 사용하도록 수정
 - 원인: 코드 리뷰 시 가독성이 안좋음.
 - 해결: tab 대신 space 로 사용토록 수정

SangYunLeee committed:
Add: 자기 자신 회사의 게시글 가져오는 API 추가

SangYunLeee committed:
fix: 서버사이드 build Error 수정

SangYunLeee committed:
fix: 더이상 유효하지 않는 주석 제거
